events/service/src/main/java/org/projectnessie/events/service/EventSubscribers.java

```diff
@@ -119,21 +107,23 @@ public synchronized void start(Function<EventSubscriber, EventSubscription> subs
   @Override
   public synchronized void close() {
     if (!closed) {
-      LOGGER.info("Closing subscribers...");
-      List<Exception> errors = new ArrayList<>();
-      for (EventSubscriber subscriber : subscribers) {
-        try {
-          subscriber.close();
-        } catch (Exception e) {
-          errors.add(e);
+      if (!subscribers.isEmpty()) {
+        LOGGER.info("Closing subscribers...");
+        List<Exception> errors = new ArrayList<>();
+        for (EventSubscriber subscriber : subscribers) {
+          try {
+            subscriber.close();
+          } catch (Exception e) {
+            errors.add(e);
+          }
         }
+        if (!errors.isEmpty()) {
+          RuntimeException e = new RuntimeException("Error closing at least one subscriber");
+          errors.forEach(e::addSuppressed);
+          throw e;
+        }
+        LOGGER.info("Done closing subscribers.");
```

Comment:
This is unreachable in case of closing exceptions... I'm not sure where the `RuntimeException` goes... would it make sense to log it just in case?

I know this behaviour is not changed by this PR, so take this comment as a nit :)

Reply:
I fear that it would be logged twice, but why not – better twice than never ;-)